src/components/EventUserTable.js

```diff
@@ -1,54 +1,300 @@
+import { REGISTRATION_STATUS } from '../constants/Constants';
+import Select from '@material-ui/core/Select';
 import MenuItem from '@material-ui/core/MenuItem';
+import Typography from '@material-ui/core/Typography';
+import { RadialChart, XYPlot, XAxis, YAxis, VerticalGridLines, HorizontalGridLines, VerticalBarSeries } from 'react-vis';
 
+const styles = {
+  stats: {
+    width: '100%',
+    display: 'flex',
+    margin: '6px',
+    borderRadius: '20px',
+    boxShadow: 'rgba(0, 0, 0, 0.4) 0 0 10px',
+    cursor: 'pointer'
+  },
+  stat: {
+    margin: '10px'
+  }
+}
 /**
  * Class component that displays event user table populated from the backend
  * When a user check-in status is changed, the backend is updated and it fetches new data
  */
 export class EventUserTable extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.changeVisibility = this.changeVisibility.bind(this)
+    this.state = {
+      registrationObj: { temp: 0 },
+      faculties: {},
+      years: {},
+      dietary: {},
+      genders: {},
+      heardFrom: {},
+      'registrationVisible': { visible: false, style: { display: 'none' } },
+      facultyVisible: { visible: false, style: { display: 'none' } },
+      yearVisible: { visible: false, style: { display: 'none' } },
+      dietaryVisible: { visible: false, style: { display: 'none' } },
+      gendersVisible: { visible: false, style: { display: 'none' } },
+      heardFromVisible: { visible: false, style: { display: 'none' } }
+    };
   }
 
   async updateUserRegistrationStatus(id, registrationStatus) {
     const body = {
       eventID: this.props.event.id,
-      id,
       registrationStatus
     };
 
-    await fetchBackend('/registrations', 'POST', body);
+    await fetchBackend(`/registrations/${id}`, 'PUT', body);
 
     this.getEventTableData(this.props.event.id);
   }
 
+  /* updates stats and the rows in the table
+     faculty, gender, dietary, and year stats are only computed on the initial render of the component
+     # of registered/checkedin etc. is computed every single time this function is called
+  */
   async getEventTableData(eventID) {
-    const params = new URLSearchParams({
+    let params = new URLSearchParams({
+      eventID: eventID
+    })
+    await fetchBackend(`/registrations/?${params}`, 'GET')
+      .then(response => {
+        let heardFrom = {};
+        for (let i = 0; i < response.size; ++i) {
+          const temp = response.data[i];
+          if (temp.hasOwnProperty('heardFrom')) {
+            heardFrom[temp.heardFrom] = heardFrom.hasOwnProperty(temp.heardFrom) ? heardFrom[temp.heardFrom] + 1 : 1;
+          }
+        }
+        this.setState({ heardFrom })
+      })
+
+    params = new URLSearchParams({
       users: true
     });
 
-    fetchBackend(`/events/${eventID}${params}`, 'GET')
-      .then(response => {
-        this.setState({ rows: response });
+    await fetchBackend(`/events/${eventID}?${params}`, 'GET')
```

Comment:
Hm, kinda wish we didn't have to make two api calls to get this info. We should consider combining the `registrations?eventId=xxx` and `events/:id?users=true` endpoints